src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int EXPANSION_FACTOR1 = 1;
+    private static final int EXPANSION_FACTOR2 = 2;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growArrayIfRequred();
+        elements[size] = value;
+        size++;
```

Comment:
```suggestion
        elements[size++] = value;
```
not a mistake but nice to do in one line